charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -30,8 +30,9 @@ data:
           nodeSelector: {{ .Values.meshGateway.nodeSelector }}
           serviceType: {{ .Values.meshGateway.service.type }}
     meshGateways:
-      - name: mesh-gateway
-        namespace: {{ .Release.Namespace }}
+      - metadata:
+          name: mesh-gateway
+          namespace: {{ .Release.Namespace }}
```

Comment:
Big fan of this change. Thoughts on just making it the whole resource like we did with `GatewayClassConfig`? I know we can derive the `apiVersion` and `kind` from the fact that it's in this list, but the organization part of my brain prefers we be consistent anyways. It also makes slurping up the config a lil more straightforward

Reply:
can do!